### ISSUE TITLE
Add Option to Resize Floating Windows Upon Toggling Floating Mode

### ISSUE THREAD

**Issue by TrueMyst** [hyprwm/Hyprland#7926]
### Description

Right now, when you toggle floating mode in Hyprland, the window keeps its original size, and I’ve found it a bit inconvenient. It would be super helpful to have an option that automatically resizes the window to specific dimensions (like width and height) when floating mode is toggled.

At the moment, I’m using a [script]() to manage this manually:

```bash
#!/bin/bash

floating=$(hyprctl activewindow -j | jq '.floating')
window=$(hyprctl activewindow -j | jq '.initialClass' | tr -d "\"")

function toggle() {
	width=$1
	height=$2

	hyprctl --batch "dispatch togglefloating; dispatch resizeactive exact ${width} ${height}; dispatch centerwindow"
}

function untoggle() {
	hyprctl dispatch togglefloating
}

function handle() {
	width=$1
	height=$2

	if [ "$floating" == "false" ]; then
		toggle "$width" "$height"
	else
		untoggle
	fi
}

case $window in
kitty) handle "50%" "55%" ;;
*) handle "80%" "75%" ;;
esac
```
It’d be really great to have this as a built-in feature so that floating windows resize automatically with predefined values when toggled. It would make things much smoother :>

**Comment by christhepanda28** [comment]
# Kitty windows float and resize to 50% width and 55% height
windowrulev2 = float, class:^(kitty)$
windowrulev2 = size 50% 55%, class:^(kitty)$
windowrulev2 = center, class:^(kitty)$

# Other windows float and resize to 80% width and 75% height
windowrulev2 = float, class:^(.*)$
windowrulev2 = size 80% 75%, class:^(.*)$
windowrulev2 = center, class:^(.*)$

**Comment by TrueMyst** [comment]
But the problem with this solution is that, it floats the window by default when you open them.

**Comment by christhepanda28** [comment]
ah, should have made it check for floating, not make everything float

``` 
windowrulev2 = size 50% 55%, class:^(kitty)$,floating: 1
windowrulev2 = center, class:^(kitty)$, floating: 1
```
```
windowrulev2 = size 80% 75%, class:^(.*)$, floating: 1
windowrulev2 = center, class:^(.*)$, floating: 1
```

**Comment by TrueMyst** [comment]
Haha, the problem still occurs :>

**Comment by christhepanda28** [comment]
that's weird when I test it on my end that isn't the case. did you get rid of these lines?
```
windowrulev2 = float, class:^(.*)$
windowrulev2 = float, class:^(kitty)$
```

**Comment by TrueMyst** [comment]
No I didn't

**Comment by christhepanda28** [comment]
When you do it should work as expected

Window rule wiki page has extra info if you want to extend in future

**Comment by TrueMyst** [comment]
> When you do it should work as expected
> 
> Window rule wiki page has extra info if you want to extend in future

Hmm weird, it doesn't seem to work for me

**Comment by fangjh13** [comment]
Yes I have this problem too, I just want to set the size when it becomes floating, not when it opens in floating mode, and only configure it for a separate application such as kitty, is there any other way to solve it besides using scripts?

**Comment by vaxerski** [comment]
Hello there!

This issue has been closed, as we are moving from issues to discussions. More about this can be found [here](https://wiki.hyprland.org/Contributing-and-Debugging/Issue-Guidelines/) and in #9854.

Firstly, please make sure this issue is still relevant in latest Hyprland. If so, we ask you to open a [discussion](https://github.com/hyprwm/Hyprland/discussions) (please read the discussion guidelines first, in the pinned post)

Thank you for your understanding!

**Comment by au-summer** [comment]
A hack solution will be to write a script instead of directly call togglefloating. For example:

```
# get the class and floating status of the active window
active_class=$(hyprctl activewindow | awk -F": " '/class:/ {print $2}')
is_floating=$(hyprctl activewindow | awk -F": " '/floating:/ {print $2}')

# if the window is already floating, toggle it back to tiled
if [[ $is_floating == "1" ]]; then
    hyprctl dispatch togglefloating
    exit 0
fi

# resize and float the active window based on its class
if [[ $active_class == *"firefox"* ]]; then
    hyprctl dispatch togglefloating
    hyprctl dispatch centerwindow
    hyprctl dispatch resizeactive exact 50% 100%
else
    hyprctl dispatch togglefloating
fi

```

And you call it like `bind = $mainMod, W, exec, $scrPath/togglefloating.sh`.

**Comment by adriancosta21** [comment]
This solution worked really well! But I've noticed that the centerwindow function wasn't always working. Specifically, it only worked when I toggled the same window to floating more than once.

To solve this,  you just have to  center the window **after** resizing it:

    hyprctl dispatch resizeactive exact 50% 100%
    hyprctl dispatch centerwindow
Now the windows will always be centered when toggling. Thank you!

**Comment by adminy** [comment]
the issue I'm having is that I spend a lot of time resizing and moving windows around ... then on restart its just defaulting to weird sizes instead of remembering what I do to them. there is no state I guess ... my config is mostly just `float true, match:class ^(.*)$`

**Comment by Jay-N7** [comment]
Hey how can we implement this in the new lua hyprland config? Will it be a window rule or part of the key-bind for toggling float?
Thanks
